protocol-units/da/m1/light-node/src/v1/manager.rs

```diff
@@ -12,7 +24,14 @@ where
 }
 
 // Implements a very simple manager using a marker strategy pattern.
-impl Manager<LightNodeV1> {
+impl<C> Manager<LightNodeV1<C>>
+where
+	C: PrimeCurve + CurveArithmetic + DigestPrimitive + PointCompression,
+	Scalar<C>: Invert<Output = CtOption<Scalar<C>>> + SignPrimitive<C>,
+	SignatureSize<C>: ArrayLength<u8>,
+	AffinePoint<C>: FromEncodedPoint<C> + ToEncodedPoint<C> + VerifyPrimitive<C>,
+	FieldBytesSize<C>: ModulusSize,
```

Comment:
Perhaps it would be good to add a helper trait that represents these lengthy bounds.

Reply:
Was going to call it `SigningCurve` btw.

Reply:
Yeah, I think it's the associated types that are making this tricky, but here are all of the variants I have tried...

```rust
pub trait SigningCurve: PrimeCurve + CurveArithmetic + DigestPrimitive + PointCompression
where
	SignatureSize<Self>: ArrayLength<u8>,
	AffinePoint<Self>: FromEncodedPoint<Self> + ToEncodedPoint<Self> + VerifyPrimitive<Self>,
	FieldBytesSize<Self>: ModulusSize,
{
	// You can define other methods or associated types specific to curve operations
}

pub trait SigningKeyRequirements<C: SigningCurve>
where
	Scalar<C>: Invert<Output = CtOption<Scalar<C>>> + SignPrimitive<C>,
{
	// You can define any signing-specific logic here if needed
}
```

```rust
pub trait SigningCurve: PrimeCurve + CurveArithmetic + DigestPrimitive + PointCompression
where
	Scalar<Self>: Invert<Output = CtOption<Scalar<Self>>> + SignPrimitive<Self>,
	SignatureSize<Self>: ArrayLength<u8>,
	AffinePoint<Self>: FromEncodedPoint<Self> + ToEncodedPoint<Self> + VerifyPrimitive<Self>,
	FieldBytesSize<Self>: ModulusSize,
{
}
```

```rust
pub trait SigningCurve<C> :
where
	C: PrimeCurve + CurveArithmetic + DigestPrimitive + PointCompression,
	Scalar<C>: Invert<Output = CtOption<Scalar<C>>> + SignPrimitive<C>,
	SignatureSize<C>: ArrayLength<u8>,
	AffinePoint<C>: FromEncodedPoint<C> + ToEncodedPoint<C> + VerifyPrimitive<C>,
	FieldBytesSize<C>: ModulusSize,
{
}
```

Reply:
Yeah, I've played with the second variant and did not get far.

The `ecdsa` API is ungodly abstract, could we achieve the same with [ed25519-dalek](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/struct.SigningKey.html)?

Reply:
Maybe just replace the generic parameter with the [specific curve](https://docs.rs/k256/latest/k256/struct.Secp256k1.html), and get into all this abstract bound crap when we actually need it.